pkg/testcase/privateregistry.go

```diff
@@ -0,0 +1,96 @@
+package testcase
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/rancher/distros-test-framework/pkg/customflag"
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/gomega"
+)
+
+var token string
+
+func TestBuildPrivateCluster(cluster *shared.Cluster) {
+	Expect(cluster.Status).To(Equal("cluster created"))
+	Expect(cluster.ServerIPs).ShouldNot(BeEmpty())
+
+	if cluster.BastionConfig.PublicIPv4Addr != "" {
+		shared.LogLevel("info", "Bastion Public IP: "+cluster.BastionConfig.PublicIPv4Addr)
+		shared.LogLevel("info", "Bastion Public DNS: "+cluster.BastionConfig.PublicDNS)
+	}
+	shared.LogLevel("info", "Server Node IPs: %v", cluster.ServerIPs)
+
+	checkAndPrintAgentNodeIPs(cluster.NumAgents, cluster.AgentIPs, false)
+
+	if cluster.Config.Product == "rke2" {
+		checkAndPrintAgentNodeIPs(cluster.NumWinAgents, cluster.WinAgentIPs, true)
+	}
+}
+
+func TestPrivateRegistry(cluster *shared.Cluster, flags *customflag.FlagConfig) {
+	shared.SetupPrivateRegistry(cluster, flags)
+	shared.LogLevel("info", "Copying assets on all the nodes...")
+	shared.CopyAssetsOnNodes(cluster)
+	installOnServers(cluster)
+	installOnAgents(cluster)
+}
+
+func installOnServers(cluster *shared.Cluster) {
+	serverFlags := os.Getenv("server_flags")
+	for idx, serverIP := range cluster.ServerIPs {
+		if idx == 0 {
```

Comment:
// mind adding a comment here on what we are doing?
		if idx == 0 {
			shared.LogLevel("info", "Installing %v on server node-1...", cluster.Config.Product)
			cmd := fmt.Sprintf(
				"sudo chmod +x install_product.sh; "+
					`sudo ./install_product.sh "%v" "" "" "server" "%v" "%v"`,
				cluster.Config.Product, serverIP, serverFlags)
			_, err := shared.CmdForPrivateNode(cluster, cmd, serverIP)
			Expect(err).To(BeNil())

			cmd = fmt.Sprintf("sudo cat /var/lib/rancher/%v/server/token", cluster.Config.Product)
			token, err = shared.CmdForPrivateNode(cluster, cmd, serverIP)
			Expect(err).To(BeNil())
			Expect(token).NotTo(BeEmpty())
			shared.LogLevel("debug", "token: "+token)
		}

		// mind adding a comment here on what we are doing?
		if idx > 0 {
			shared.LogLevel("info", "Installing %v on server node-%v...", cluster.Config.Product, idx+1)
			cmd := fmt.Sprintf(
				"sudo chmod +x install_product.sh; "+
					`sudo ./install_product.sh "%v" "%v" "%v" "server" "%v" "%v"`,
				cluster.Config.Product, cluster.ServerIPs[0], token, serverIP, serverFlags)
			_, err := shared.CmdForPrivateNode(cluster, cmd, serverIP)
			Expect(err).To(BeNil())
		}
	}
}

Reply:
Sure will add